rust/rbac-registration/src/cardano/cip509/mod.rs

```diff
@@ -179,16 +214,14 @@ impl Cip509 {
         let mut is_valid_stake_public_key = true;
         let mut is_valid_payment_key = true;
         let mut is_valid_signing_key = true;
-        if let Some(role_set) = &self.x509_chunks.0.role_set {
-            // Validate only role 0
-            for role in role_set {
-                if role.role_number == 0 {
-                    is_valid_stake_public_key =
-                        validate_stake_public_key(self, txn, validation_report).unwrap_or(false);
-                    is_valid_payment_key =
-                        validate_payment_key(txn, role, validation_report).unwrap_or(false);
-                    is_valid_signing_key = validate_role_singing_key(role, validation_report);
-                }
+        // Validate only role 0
+        for role in &self.metadata.role_set {
```

Comment:
Should not need to iterate here, should be able to reference it directly by role number.
This is a very common behaviour.  the Cip509 struct should have a method to return the roles it defines, and then to query information from any role it contains.

conceptually:
```
self.metadata.role_set.get(0)
```
should return `Some` data if its in the registration or `None` if its not.
And we should be able to do something like:
```
self.metadata.role_set.all_roles() -> Vec<RoleNumber>
```
To get a list of all Role Values defined in the registration.

As said above, we should use this internally, and it should also be exposed public API for a consumer.